c/datatable.cc

```diff
@@ -89,17 +106,17 @@ size_t DataTable::xcolindex(int64_t index) const {
 /**
  * Make a shallow copy of the current DataTable.
  */
-DataTable* DataTable::copy() const {
-  colvec newcols = columns;  // copy the vector
-  DataTable* res = new DataTable(std::move(newcols), this);
-  res->nkeys = nkeys;
+DataTable DataTable::copy() const {
```

Comment:
What about renaming this to `clone()` for consistency with the corresponding `ColumnImpl` method? Btw, why do you think `clone` is a better name than `shallowcopy`?

Reply:
Some time long ago we had `shallowcopy()` (which copied a reference to Column's buffer), and `deepcopy()`, which copied the buffer's content.
Today, `deepcopy()` no longer exists -- the buffer is copied automatically and only when necessary. But we added a "standard" copy too: `Column(const Column&)` which is even "shallower" than `shallowcopy()`. Thus, in order to avoid possible misunderstandings, I renamed `shallowcopy()` into `clone()`.

The `DataTable::copy()` method could be renamed into `clone()` as well, I don't see any contradiction there. Right now, however, its name mimics python's `Frame.copy()` (which in turns follows standard python naming conventions).

Reply:
On a second thought, we don't have to name `DataTable::copy()` at all: this could be just a regular copy-constructor `DataTable(const DataTable&)`.